--- a/engine/class_modules/monk/sc_monk.cpp
+++ b/engine/class_modules/monk/sc_monk.cpp
@@ -5437,7 +5437,7 @@
 
 bool monk_t::validate_actor()
 {
-  if ( specialization() == MONK_MISTWEAVER )
+  if ( specialization() == MONK_MISTWEAVER && !sim->allow_experimental_specializations )
   {
     if ( !quiet )
       sim->error( "Mistweaver Monk for {} is not currently supported.", *this );
@@ -5485,6 +5485,7 @@
   {
     case MONK_BREWMASTER:
     case MONK_WINDWALKER:
+    case MONK_MISTWEAVER:
       return true;
     default:
       sim->error( "No specialization was selected for {}.", *this );
